File: frontend/src/types/account.ts
import type { OrderStatus } from "./admin";
import type { ContactMethod, DeliveryMethod } from "./order";

export interface ClientSession {
  id: number;
  email: string;
  phone: string | null;
}

export interface RegisterInput {
  email: string;
  password: string;
  phone?: string;
}

export interface LoginInput {
  email: string;
  password: string;
}

export interface WishlistProduct {
  product_id: number;
  slug: string;
  name: string;
  price: number;
  old_price: number | null;
  main_image: string | null;
}

export interface AddressOut {
  id: number;
  recipient_first_name: string;
  recipient_last_name: string;
  city: string;
  delivery_method: DeliveryMethod;
  np_office: string | null;
  is_default: boolean;
}

export interface AddressInput {
  recipient_first_name: string;
  recipient_last_name: string;
  city: string;
  delivery_method: DeliveryMethod;
  np_office?: string;
  is_default?: boolean;
}

export interface AccountOrderListItem {
  id: number;
  status: OrderStatus;
  total: number;
  created_at: string;
}

export interface AccountOrderListResponse {
  items: AccountOrderListItem[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}

export interface AccountOrderItem {
  id: number;
  product_id: number;
  product_name: string;
  quantity: number;
  price_at_order: number;
}

export interface AccountOrderDetail {
  id: number;
  first_name: string;
  last_name: string;
  phone: string;
  city: string;
  delivery_method: DeliveryMethod;
  np_office: string | null;
  contact_method: ContactMethod;
  comment: string | null;
  status: OrderStatus;
  promo_code: string | null;
  items_total: number;
  discount_amount: number;
  total: number;
  created_at: string;
  items: AccountOrderItem[];
}

File: backend/app/schemas/account.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, field_validator

from app.core.constants import ContactMethod, DeliveryMethod, OrderStatus


class WishlistItemOut(BaseModel):
    product_id: int
    slug: str
    name: str
    price: float
    old_price: float | None
    main_image: str | None


class WishlistAddRequest(BaseModel):
    product_id: int


class AddressOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    recipient_first_name: str
    recipient_last_name: str
    city: str
    delivery_method: DeliveryMethod
    np_office: str | None
    is_default: bool


class AddressInput(BaseModel):
    recipient_first_name: str
    recipient_last_name: str
    city: str
    delivery_method: DeliveryMethod
    np_office: str | None = None
    is_default: bool = False

    @field_validator("recipient_first_name", "recipient_last_name", "city")
    @classmethod
    def not_blank(cls, value: str) -> str:
        value = value.strip()
        if not value:
            raise ValueError("Поле обов'язкове")
        return value


class AccountOrderListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    status: OrderStatus
    total: float
    created_at: datetime


class AccountOrderListResponse(BaseModel):
    items: list[AccountOrderListItem]
    total: int
    page: int
    page_size: int
    pages: int


class AccountOrderItemOut(BaseModel):
    id: int
    product_id: int
    product_name: str
    quantity: int
    price_at_order: float


class AccountOrderDetail(BaseModel):
    id: int
    first_name: str
    last_name: str
    phone: str
    city: str
    delivery_method: DeliveryMethod
    np_office: str | None
    contact_method: ContactMethod
    comment: str | None
    status: OrderStatus
    promo_code: str | None
    items_total: float
    discount_amount: float
    total: float
    created_at: datetime
    items: list[AccountOrderItemOut]
